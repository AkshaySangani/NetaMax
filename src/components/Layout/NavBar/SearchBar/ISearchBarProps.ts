import { ISearchDropDownProps } from "./SearchDropDown/ISearchDropDownProps";
import { ISearchInputProps } from "./SearchInput/ISearchInputProps";

export interface ISearchBarProps extends ISearchDropDownProps, ISearchInputProps {}
