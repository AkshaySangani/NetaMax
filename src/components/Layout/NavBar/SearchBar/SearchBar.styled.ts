import styled, { createGlobalStyle } from "styled-components";
import { responsive } from "styled/responsive";

import { Box } from "@chakra-ui/react";

export const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
  }
`;

export const SearchBarContainer = styled(Box)`
  display: flex;
  flex-direction: column;
  width: 100%;

  ${responsive.smPlus} {
    width: 100%;
  }
`;
