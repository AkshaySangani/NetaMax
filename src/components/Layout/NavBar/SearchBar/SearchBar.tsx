import { useEffect, useRef, FC, ReactElement } from "react";

import { ISearchBarProps } from "./ISearchBarProps";
import { GlobalStyle, SearchBarContainer } from "./SearchBar.styled";
import { SearchDropDown } from "./SearchDropDown/SearchDropDown";
import { SearchInput } from "./SearchInput/SearchInput";

/**
 * The SearchBar Component
 * @param {ISearchBarProps} props The component props
 * @returns {ReactElement} the searchBar component.
 */
export const SearchBar: FC<ISearchBarProps> = (props: ISearchBarProps): ReactElement => {
  const {
    value,
    products,
    onAddToBasketClick,
    getQtyInBasket,
    onChange,
    onFocusIn,
    onFocusOut,
    onRemoveFromBasketClick,
    isLoadingSearchProduct,
    clearSearchProducts,
    handleSearchProducts,
    emptySearchMessage,
    hasMoreProducts,
  } = props;

  const myRef = useRef<HTMLDivElement>(null);

  /**
   * Handle focusout
   * @param {any} e event
   * @returns {any} void
   */
  const handleClickOutside = (e: MouseEvent) => {
    if (!myRef?.current?.contains(e.target as Node)) {
      onFocusOut && onFocusOut(e as unknown as React.FocusEvent<HTMLInputElement>);
    }
  };

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => document.removeEventListener("mousedown", handleClickOutside);
  });

  return (
    <SearchBarContainer ref={myRef}>
      <GlobalStyle />
      <SearchInput
        value={value}
        onChange={onChange}
        onFocusIn={onFocusIn}
        clearSearchProducts={clearSearchProducts}
      />
      <SearchDropDown
        value={value}
        clearSearchProducts={clearSearchProducts}
        products={products}
        onAddToBasketClick={onAddToBasketClick}
        onRemoveFromBasketClick={onRemoveFromBasketClick}
        getQtyInBasket={getQtyInBasket}
        isLoadingSearchProduct={isLoadingSearchProduct}
        handleSearchProducts={handleSearchProducts}
        emptySearchMessage={emptySearchMessage}
        hasMoreProducts={hasMoreProducts}
      />
    </SearchBarContainer>
  );
};
