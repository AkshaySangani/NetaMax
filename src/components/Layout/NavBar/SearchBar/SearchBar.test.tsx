import { render, cleanup } from "@testing-library/react";
import { SearchBar } from "./SearchBar";
import { ISearchBarProps } from "./ISearchBarProps";

describe("Render SearchBar component", () => {
  let mock: ISearchBarProps;

  beforeEach(() => {
    mock = {
      value: "Search",
      products: [],
      onAddToBasketClick: jest.fn(),
      getQtyInBasket: jest.fn(),
      onChange: jest.fn(),
      onFocusIn: jest.fn(),
      onFocusOut: jest.fn(),
      onRemoveFromBasketClick: jest.fn(),
      isLoadingSearchProduct: false,
      clearSearchProducts: jest.fn(),
      handleSearchProducts: jest.fn(),
      emptySearchMessage: "Empty search message",
      hasMoreProducts: false,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("SearchBar should be render correctly", () => {
    const { container } = render(<SearchBar {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
