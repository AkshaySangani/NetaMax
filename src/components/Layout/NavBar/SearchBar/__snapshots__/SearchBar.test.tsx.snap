// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SearchBar component SearchBar should be render correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL cWETYv css-0"
  >
    <span
      class="sc-dkPtRN hdfwtk"
    >
      <div
        class="chakra-input__group css-4302v8"
      >
        <div
          class="chakra-input__right-element css-194onc0"
          data-testid="clear button"
        >
          <svg
            fill="#3A3A3A"
            height="14px"
            viewBox="0 0 22 22"
            width="14px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.9562 11.0329C12.9351 11.0119 12.9184 10.9869 12.907 10.9594C12.8955 10.9318 12.8897 10.9023 12.8897 10.8726C12.8897 10.8428 12.8955 10.8133 12.907 10.7857C12.9184 10.7582 12.9351 10.7332 12.9562 10.7122L21.3487 2.32062C21.6035 2.06543 21.7465 1.71946 21.7463 1.35883C21.746 0.998188 21.6025 0.652422 21.3473 0.397593C21.0921 0.142764 20.7462 -0.000254524 20.3855 3.40049e-07C20.0249 0.000255204 19.6791 0.143762 19.4243 0.398952L11.0327 8.78689C11.0117 8.80798 10.9867 8.82472 10.9592 8.83614C10.9317 8.84756 10.9022 8.85344 10.8724 8.85344C10.8426 8.85344 10.8131 8.84756 10.7855 8.83614C10.758 8.82472 10.733 8.80798 10.712 8.78689L2.32043 0.398952C2.19426 0.272714 2.04445 0.172565 1.87956 0.104222C1.71468 0.03588 1.53795 0.000683039 1.35947 0.000640973C0.998998 0.000556017 0.653261 0.14367 0.398311 0.398499C0.143362 0.653328 8.49937e-05 0.998999 3.78023e-08 1.35947C-8.49181e-05 1.71993 0.143029 2.06567 0.397858 2.32062L8.78851 10.7122C8.80961 10.7332 8.82634 10.7582 8.83776 10.7857C8.84918 10.8133 8.85506 10.8428 8.85506 10.8726C8.85506 10.9023 8.84918 10.9318 8.83776 10.9594C8.82634 10.9869 8.80961 11.0119 8.78851 11.0329L0.397858 19.4254C0.27168 19.5516 0.171601 19.7015 0.103336 19.8664C0.0350717 20.0313 -4.20281e-05 20.2081 3.78023e-08 20.3865C8.49937e-05 20.747 0.143362 21.0927 0.398311 21.3475C0.524549 21.4737 0.674404 21.5738 0.839319 21.642C1.00423 21.7103 1.18098 21.7454 1.35947 21.7454C1.71993 21.7453 2.0656 21.602 2.32043 21.3471L10.712 12.9546C10.733 12.9335 10.758 12.9168 10.7855 12.9053C10.8131 12.8939 10.8426 12.888 10.8724 12.888C10.9022 12.888 10.9317 12.8939 10.9592 12.9053C10.9867 12.9168 11.0117 12.9335 11.0327 12.9546L19.4243 21.3471C19.6791 21.602 20.0248 21.7453 20.3853 21.7454C20.7457 21.7454 21.0915 21.6023 21.3464 21.3475C21.6014 21.0927 21.7446 20.747 21.7447 20.3865C21.7448 20.0261 21.6017 19.6803 21.3469 19.4254L12.9562 11.0329Z"
              fill="#3A3A3A"
              height="14px"
              width="14px"
            />
          </svg>
        </div>
        <input
          class="chakra-input css-m14b0g"
          data-testid="search-input"
          placeholder="QuÃ© antojo tienes hoy?"
          value="Search"
        />
        <div
          class="chakra-input__left-element css-o84xzr"
        >
          <svg
            fill="none"
            height="18"
            viewBox="0 0 18 18"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M11.2924 12.4458C8.65021 14.5685 4.7826 14.4014 2.33331 11.9446C-0.294696 9.30856 -0.294696 5.0347 2.33331 2.39866C4.96131 -0.237379 9.22214 -0.237379 11.8501 2.39866C14.2994 4.85544 14.466 8.73487 12.3498 11.3851L17.4897 16.5408C17.7817 16.8337 17.7817 17.3085 17.4897 17.6014C17.1977 17.8943 16.7243 17.8943 16.4323 17.6014L11.2924 12.4458ZM3.39073 10.8839C1.34673 8.83369 1.34673 5.50957 3.39073 3.45932C5.43473 1.40907 8.74872 1.40907 10.7927 3.45932C12.8352 5.50807 12.8367 8.82881 10.7972 10.8794C10.7957 10.8809 10.7942 10.8824 10.7927 10.8839C10.7912 10.8854 10.7897 10.8869 10.7882 10.8884C8.74386 12.9342 5.43324 12.9327 3.39073 10.8839Z"
              fill="#3A3A3A"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </span>
    <div
      class="sc-gsDKAQ gCkrae css-0"
    >
      <div
        class="css-na8a98"
      >
        <p
          class="chakra-text css-0"
        >
          Empty search message
           ðŸ˜”
        </p>
      </div>
    </div>
  </div>
</div>
`;
